[Huzaifa Raghav] iP

leonidas1712 opened the pull request:
# JARVIS
> "I am Iron Man" - Iron Man

JARVIS is a chatbot that gives you a CLI interface for managing tasks. Perfect for fast typers!

**Features**:
1. Three types of tasks - Todo, Deadline and Event
2. Listing, adding and deleting tasks
3. Marking and unmarking tasks as done
4. Finding tasks by keyword

**Full list of commands**:
- list
- todo /d <description>
- event /d <description> /at <date>
- deadline /d <description> /by <datetime>
- mark /id <id>
- unmark /id <id>
- delete /id <id>
- find /s <query>
- bye - exit command

### TODO:
- [ ] Help command to list commands and arguments/descriptions for each 🆘 
- [ ] Add saving of tasks upon ungraceful exit 💯 

```java
public static void main(String[] args) {
    System.out.println("hello");
}
```

Click [me](https://www.youtube.com/watch?v=dQw4w9WgXcQ) to find out more about JARVIS!

Commit:
Add Gradle support

leonidas1712 committed:
Add skeletal (echobot) JARVIS

leonidas1712 committed:
Write Input class for intermediate representation of String input

leonidas1712 committed:
Set up Command and CommandResponse classes

leonidas1712 committed:
Add classes for Tasks and TaskModel

leonidas1712 committed:
Add methods for task creation and view all to TaskModel

leonidas1712 committed:
Add support for task creation and listing

leonidas1712 committed:
Clean up TaskModel

leonidas1712 committed:
Add classes for Argument and TaskIdArgument

leonidas1712 committed:
Add support for MarkCommand and TaskIdArgument validation

leonidas1712 committed:
Refactor to use CommandRunner class for setting up commands

leonidas1712 committed:
Refactor output to use OutputLogger

leonidas1712 committed:
Add numberedOutput method for OutputLogger

leonidas1712 committed:
Refactor to use OutputLogger errorOutput

leonidas1712 committed:
Add unmark command

leonidas1712 committed:
Add description argument for TodoCommand, standardise TaskResponse format

leonidas1712 committed:
Add support for Deadline command

leonidas1712 committed:
Refactor string arguments to use StringArgument class (will need to change for later levels)

leonidas1712 committed:
Add support for event command

leonidas1712 committed:
Add comments, configure testing

leonidas1712 committed:
Fix testing issues

leonidas1712 committed:
Add Javadoc desc for command classes

leonidas1712 committed:
Move packages inside java folder and update runtest.sh accordingly

leonidas1712 committed:
Move packages to java folder

leonidas1712 committed:
Refactor to use custom DukeException class

leonidas1712 committed:
Add getTask method for TaskModel to standardise error msg upon invalid index

leonidas1712 committed:
Add DeleteCommand and update test files accordingly

leonidas1712 committed:
Move Duke to entry package and update runtest.sh

leonidas1712 committed:
Move Duke.java to entry package

leonidas1712 committed:
Merge branch 'add-gradle-support'

leonidas1712 committed:
Add support for Gradle checkstyle

leonidas1712 committed:
Fix style for Input.java

leonidas1712 committed:
Fix style for arguments package

leonidas1712 committed:
Fix style for commands package

leonidas1712 committed:
Fix style for Duke.java

leonidas1712 committed:
Fix style for OutputLogger

leonidas1712 committed:
Fix style for task package

leonidas1712 committed:
Add Storage class

leonidas1712 committed:
Update build.gradle

leonidas1712 committed:
Merge branch 'master' into branch-Level-7

leonidas1712 committed:
Fix issue with CommandRunner reset

leonidas1712 committed:
Fix Storage.java init

leonidas1712 committed:
Update runtest.sh to delete data folder inside

leonidas1712 committed:
Add TaskType enum for serialization purposes

leonidas1712 committed:
Add org.json dependency

leonidas1712 committed:
Add saving of tasks to text file using JSON

leonidas1712 committed:
Add loading of existing tasks

leonidas1712 committed:
Bugfix: mark as done not stored to text

leonidas1712 committed:
Add DatetimeArgument class

leonidas1712 committed:
Add datetime argument support for Deadline

leonidas1712 committed:
Merge branch 'branch-Level-7'
This branch adds support for saving and loading tasks from disk using text file

leonidas1712 committed:
Merge branch 'branch-Level-8'

Merge changes for datetime and saving to file

leonidas1712 committed:
Fix style for TaskDeserializer

leonidas1712 committed:
Fix style for DatetimeArgument

leonidas1712 committed:
Fix style for DeadlineCommand

leonidas1712 committed:
Add JUnit tests for Input.java

leonidas1712 committed:
Add tests for StringArgument

leonidas1712 committed:
Refactor StringArgumentTest to useInputStub

leonidas1712 committed:
Fix style for Input.java

leonidas1712 committed:
Add Javadoc

leonidas1712 committed:
Add JavaDoc for test files

leonidas1712 committed:
Fix this references for coding standard

leonidas1712 committed:
Add find command

leonidas1712 committed:
Merge branch 'branch-A-CodingStandard'
Merge coding standard branch

leonidas1712 committed:
Merge branch 'branch-Level-9'
Merge Level-9 branch implementing Find

leonidas1712 committed:
Fix style for FindCommand

tantzeyeong commented on src/test/java/input/InputTest.java:
```diff
@@ -0,0 +1,60 @@
+
+import exceptions.DukeException;
+
+
+
+
```
Think these additional blank lines were unintentionally left there? Saw this in some of the other files too.

leonidas1712 replied:
Thanks for the feedback!

tantzeyeong commented on src/test/java/arguments/StringArgumentTest.java:
```diff
@@ -0,0 +1,56 @@
+            Input invalid = new InputStub("deadline /d");
+            StringArgument s = new StringArgument(invalid, "d", "Empty", "Missing");
+            s.validate();
+        } catch (DukeException e) {
```
Perhaps naming the variable "e" with a longer word like "exception" may be more elegant? Noticed it was named differently, such as "err", in other try-catch blocks too

tantzeyeong commented on src/main/java/task/Todo.java:
```diff
@@ -0,0 +1,20 @@
+/**
+ * Class representing a Todo task
+ */
+public class Todo extends Task {
```
"To do" contains two words, so perhaps it's better to name the class "ToDo" instead?

leonidas1712 committed:
Add JavaFX to build.gradle

tantzeyeong reviewed (commented):
I feel that your code in general is very clean and well organised! I couldn't find much coding standard violations at all besides the minor ones in my comments. Really like your chatbot's name too! :)

leonidas1712 committed:
Add GUI files

leonidas1712 committed:
Integrate GUI with Jarvis classs

leonidas1712 committed:
Fix style issues

leonidas1712 committed:
Add checkstyle to Intellij

leonidas1712 committed:
Add variable arguments

leonidas1712 committed:
Merge branch 'A-CheckStyle'

leonidas1712 committed:
Add Java assertions

Codebase did not use Java assertions before.

Java assertions are commonly used to document important assumptions that
hold at important points in the code.

Let's add assertions to important points in the code.

leonidas1712 committed:
Add some codestyle changes

leonidas1712 committed:
Improve code quality

Code is not examined strictly for code quality guidelines.

Let's change the code to adhere to code quality guidelines through
changes such as keeping methods short and deleting dead code.

leonidas1712 committed:
Remove redundant local variable

leonidas1712 committed:
Merge pull request #1 from huzaifa1712/branch-A-Assertions

Add Java assertions

leonidas1712 committed:
Merge branch 'master' into branch-A-CodeQuality

leonidas1712 committed:
Merge branch 'master' into branch-A-Streams

branch-A-Streams does not have changes made to master branch.

Let's merge changes from the master branch into this branch.

leonidas1712 committed:
Merge pull request #2 from huzaifa1712/branch-A-CodeQuality

Branch a code quality

leonidas1712 committed:
Merge branch 'master' into branch-A-Streams

leonidas1712 committed:
Merge pull request #3 from huzaifa1712/branch-A-Streams

Branch a streams

leonidas1712 committed:
Remove unused code

leonidas1712 committed:
Add CI

leonidas1712 committed:
Merge pull request #4 from huzaifa1712/A-CI

CI support

leonidas1712 committed:
Set up help command

leonidas1712 committed:
Add method to strip tokens of empty strings

leonidas1712 committed:
Add help messages for arguments

leonidas1712 committed:
Add support for help command

Jarvis does not support viewing the available commands or help messages
for them.

Let's add a help command to support this functionality.

leonidas1712 committed:
Fix checkstyle issues

leonidas1712 committed:
Merge pull request #5 from huzaifa1712/C-Help

C-Help

leonidas1712 committed:
Remove unused shortDescription methods

leonidas1712 committed:
Add CSS for dialog color and radius

leonidas1712 committed:
Add displayPicture CSS

leonidas1712 committed:
Add rounded images

leonidas1712 committed:
Fix scroll issue

When messages overflow the height of ScrollPane, the window
automatically shrinks in width unnecessarily.

Let's fix this behaviour by changing the hbarPolicy in MainWindow.fxml.

leonidas1712 committed:
Change width of ImageView

leonidas1712 committed:
Fix cut off issue for labels upon scroll

leonidas1712 committed:
Add different coloring for user and bot dialogs

leonidas1712 committed:
Add JavaDoc comments

leonidas1712 committed:
Add documentation

leonidas1712 committed:
Add acknowledgements

leonidas1712 committed:
Remove javafx18 from fxml files
